Mobile-sdk.mdx (#108)
devrev/fern-api-docs: fern/docs/pages/plug-sdk/sdk-ios.mdx

Reviewer: There were a few minor changes in the README in devrev/ios-sdk-source#89.

Author: Updated

```swift
DevRev.identifyAnonymousUser(userID:)
```

### Unverified identification

The unverified identification method identifies users with a unique identifier, but it does not verify their identity with the DevRev backend.

```swift
DevRev.identifyUnverifiedUser(_:)
```

The function accepts the `DevRev.Identity` structure, where the user identifier (`userId`) is the only required property; all other properties are optional.

### Update the user

You can update the user's information using the following method:

```swift

DevRev.updateUser(_:)

```

This function accepts the `DevRev.Identity` structure.

<Callout intent="note">
The identification functions are asynchronous. Ensure you wrap them in a `Task` when calling from synchronous contexts.
</Callout>

For example:

```swift
// Identify an anonymous user without a user identifier.
await DevRev.identifyAnonymousUser()

// Identify an unverified user using their email address as the user identifier.
await DevRev.identifyUnverifiedUser(Identity(userID: "foo@example.org"))

// Update the user's information.
await DevRev.updateUser(Identity(organizationID: "foo-bar-1337"))
```

## PLuG support chat

### UIKit

> The support chat feature can be displayed either as a modal screen from a specific view controller or pushed onto a navigation stack. 

Author: @bojan Regarding "we added another method of how the support chat can be shown on screen." I have checked the Readme file and unable to find any newly added method under UIKit section, please provide the info or tag me in the Readme file.

While the auto-masking feature may be sufficient for most situations, you can manually mark additional views as sensitive using the following method:

```swift
DevRev.markSensitiveViews(_:)
```

If any previously masked views need to be unmasked, you can use the following method:

```swift
DevRev.unmarkSensitiveViews(_:)
```

### Timers

The DevRev SDK offers a timer mechanism to measure the time spent on specific tasks, allowing you to track events such as response time, loading time, or any other duration-based metrics.

The mechanism utilizes balanced start and stop methods, both of which accept a timer name and an optional dictionary of properties.

To start a timer, use the following method:

```swift
DevRev.startTimer(_:properties:)
```

To balance it with a stop method, use the following:

```swift
DevRev.stopTimer(_:properties:)
```

For example:

```swift
DevRev.startTimer("response-time", properties: ["id": "foo-bar-1337"])

// Perform the task that you want to measure.

DevRev.stopTimer("response-time", properties: ["id": "foo-bar-1337"])
```

### Screen tracking

> The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although activities are automatically tracked, you can manually track screens using the following method:

Reviewer: We need to tweak this a bit since iOS doesn't have activities:
```suggestion
The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers are automatically tracked, you can manually track screens using the following method:
```

```swift
DevRev.trackScreenName(_:)
```

For example:

```swift
DevRev.trackScreenName("profile-screen")
```

## Push notifications

You can configure your app to receive push notifications from the DevRev SDK. The SDK is designed to handle push notifications and execute actions based on the notification's content.

The DevRev backend sends push notifications to your app to alert users about new messages in the PLuG support chat.

### Configuration

> To receive push notifications, you need to configure your DevRev organization by following the Push Notifications Integration guide (**TBD ITEM**).

Reviewer: We need Ribhu's input here for the **TBD ITEM**.

> You need to ensure that your iOS app is configured to receive push notifications. You can follow the [Apple documentation](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns) for guidance on registering your app with Apple Push Notification Service (APNS).

Reviewer: Apple naming, don't ask 🤣 
```suggestion
You need to ensure that your iOS app is configured to receive push notifications. You can follow the [Apple documentation](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns) for guidance on registering your app with Apple Push Notification Service (APNs).
```

### Register for push notifications

<Callout intent="note">
Push notifications require SDK configuration and user identification, whether unverified or anonymous, to ensure delivery to the correct user.
</Callout>

The DevRev SDK offers a method to register your device for receiving push notifications. You can register for push notifications using the following method:

```swift
DevRev.registerDeviceToken(_:deviceID:)
```

> The method requires a device identifier, which can either be an identifier unique to your system or the Apple-provided Vendor Identifier (IDFV). Typically, registration is called from the `AppDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:)` method.

Reviewer: ```suggestion
The method requires a device identifier, which can either be an identifier unique to your system or the Apple-provided Vendor Identifier (IDFV). Typically, the token registration is called from the `AppDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:)` method.
```

For example:

```swift
UIApplication.shared.unregisterForRemoteNotifications()

Task {
	guard
		let deviceID = UIDevice.current.identifierForVendor?.uuidString
	else {
		return
	}

	await DevRev.unregisterDevice(deviceID)
}
```

### Handle push notifications

To properly handle push notifications, implement the following method, typically in either the `UNUserNotificationCenterDelegate.userNotificationCenter(_:didReceive:)` or `UIApplicationDelegate.application(_:didReceiveRemoteNotification:fetchCompletionHandler:)`:

```swift
DevRev.processPushNotification(_:)
```

For example:

```swift
func userNotificationCenter(
	_ center: UNUserNotificationCenter,
	didReceive response: UNNotificationResponse
) async {
	await DevRev.processPushNotification(response.notification.request.content.userInfo)
}
```

## Sample app

This repository includes a sample app demonstrating use cases for both UIKit and SwiftUI.

Before using the sample app, configure it with your Apple Developer team and DevRev credentials. For convenience, the code is marked with compiler error directives (`#error`) in areas that require your attention.

1. Add your credentials:

> - For SwiftUI, add your credentials to `ContentView.swift`.

Reviewer 2: Indent
```suggestion
   - For SwiftUI, add your credentials to `ContentView.swift`.
```
